codalab/migration.py

```diff
@@ -195,56 +252,103 @@ def delete_original_bundle(self, bundle_uuid, bundle_location):
+                    logging.info(f"Bundle {bundle_uuid} info in database is not properly updated")
+                    raise Exception(
+                        f"Bundle {bundle_uuid} info in database is not properly updated"
+                    )
+                try:
+                    
+                    self.delete_original_bundle_by_uuid(bundle_uuid, origin_bundle_location)
+                except Exception as e:
+                    # If the bundle is not deleted, save the information in the file
+                    logging.error(f"[migration] Delete Original Bundle Error: {str(e)}")
+                    f.write(line)
+            
+
 
 if __name__ == '__main__':
     # Command line parser, parse the worksheet id
     parser = argparse.ArgumentParser(
         description='Manages your local CodaLab Worksheets service deployment'
     )
+    parser.add_argument(
+        '-a', '--all', help='Run migration on all worksheets and all bundles', action='store_true',
+    )
     parser.add_argument(
         '-w', '--worksheet', type=str, help='The worksheet uuid that needs migration'
     )
-    parser.add_argument('--target_store_name', type=str, help='The destination bundle store name')
     parser.add_argument(
-        '-d',
-        '--dry-run',
-        help='Only upload the bundle to Azure, does not modify database',
-        action='store_true',
+        '-t', '--target_store_name', type=str, help='The destination bundle store name'
     )
     parser.add_argument(
-        '-k', '--keep', help='Keep bundle content in origin bundle store', action='store_true'
+        '-c', '--change_db', help='Change the bundle location in the database', action='store_true',
     )
+    parser.add_argument('-d', '--delete', help='Delete the original database', action='store_true')
 
     args = parser.parse_args()
 
     worksheet_uuid = args.worksheet
     target_store_name = (
         "azure-store-default" if args.target_store_name is None else args.target_store_name
     )
-    if worksheet_uuid is not None and not spec_util.UUID_REGEX.match(worksheet_uuid):
-        raise Exception("Input worksheet uuid has wrong format. ")
 
-    # TODO: write output to log / log files
     migration = Migration(target_store_name)
     migration.setUp()
 
-    bundle_uuids = migration.get_bundle_uuids(worksheet_uuid)
+    if args.all:
+        bundle_uuids = migration.get_bundle_uuids(worksheet_uuid=None)
+    else:
+        # Must specify worksheet uuid
+        if worksheet_uuid is not None and not spec_util.UUID_REGEX.match(worksheet_uuid):
+            raise Exception("Input worksheet uuid has wrong format. ")
+        bundle_uuids = migration.get_bundle_uuids(worksheet_uuid)
 
     for bundle_uuid in bundle_uuids:
-        logging.info(bundle_uuid)
-
         bundle = migration.get_bundle(bundle_uuid)
-        if bundle.bundle_type != 'dataset' or bundle.state != 'ready':
+
+        # TODO: change this to allow migration of run bundles
```

Comment:
It looks like this change (removing bundle_type != 'dataset') allows for migration of run bundles, now, right?

Also, I just want to be sure: not migrating non-ready bundles is OK as long as, before we start the migration, the default upload bundle store is Azure blob (as referenced [here](https://github.com/codalab/codalab-worksheets/pull/4525#issuecomment-1699860312)), right? Although we may want to make sure uploading to disk isn't even possible as well...

Reply:
> It looks like this change (removing bundle_type != 'dataset') allows for migration of run bundles, now, right?
> 

U r right.

> Also, I just want to be sure: not migrating non-ready bundles is OK as long as, before we start the migration, the default upload bundle store is Azure blob (as referenced [here](https://github.com/codalab/codalab-worksheets/pull/4525#issuecomment-1699860312)), right? Although we may want to make sure uploading to disk isn't even possible as well...

Even we set the Azure as default storage,  user could still set `--store [some-store-on-disk]` when `cl upload`, and upload to disk.